MIP/mip-16/README.md

```diff
@@ -0,0 +1,108 @@
+
+# MIP-16: Calculation of gas fees 
+
+- **Description**: How to calculate gas fees for transactions on Movement L2.
+- **Authors**: [Franck Cassez](mailto:franck.cassez@movementlabs.xyz)
+- **Desiderata**: [MIP-\<number\>](../MD/md-\<number\>)
+
+
+## Abstract
+This MIP describes the structure of the gas fees and overall resource consumption needed to execute, and post a transaction to the DA/L1.
+
+
+## Motivation
+
+The transaction fees on L2s are multi-dimensional, so it may be useful to clarify what the fees are and how they are charged to the transaction issuer.
```

Comment:
I think this MIP needs to present a basic premise on how to handle the different fees here or otherwise to set up for contrasting several approaches.

For example, here are two alternative premises:
- Transaction sender only pays fees for execution (and related work). We do not attempt to raise DA and L1 fees up to the sender. Instead these are seen as extrinsic costs which are only modeled against validator rewards.
- Transaction sender pays fees reflecting execution, DA, and L1 work. DA and L1 fees are market-based, i.e., adjusted based on validator participation.

Reply:
There is something I don't understand here: what if a user is not a validator?

In an L2 context, you usually want to cover your operating costs. So you factor in the costs of publishing some data to the DA/L1 to make sure you don't run a deficit (or not for too long). 

As a result I don't think we need the first premise, the second alternative holds. And if we want to specialise the second alternative to cover the first, we just multiple the DA/L1 by zero, which nullifies them.

Is there anything I am missing here?

Reply:
> There is something I don't understand here: what if a user is not a validator?

There are already other logical divisions of work and operations. I am using the term validator here for simplicity.

Reply:
> In an L2 context, you usually want to cover your operating costs. So you factor in the costs of publishing some data to the DA/L1 to make sure you don't run a deficit (or not for too long).

> As a result I don't think we need the first premise, the second alternative holds. And if we want to specialize the second alternative to cover the first, we just multiple the DA/L1 by zero, which nullifies them.

You still need to be resistant to an attack. If you are naive, you can wind up at the extremes: (1) you eat the costs, say by minting token excessively, and the token inflates, security degrades; or (2) you have your validators eat the costs and they eventually stop participating.